packages/maskbook/src/web3/UI/EthereumAccountButton.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback } from 'react'
+import { makeStyles, Theme, createStyles, Button, ButtonProps } from '@material-ui/core'
+import ErrorIcon from '@material-ui/icons/Error'
+import FiberManualRecordIcon from '@material-ui/icons/FiberManualRecord'
+import { useStylesExtends } from '../../components/custom-ui-helper'
+import { useWallet } from '../../plugins/Wallet/hooks/useWallet'
+import { ProviderIcon } from '../../components/shared/ProviderIcon'
+import { formatEthereumAddress } from '../../plugins/Wallet/formatter'
+import { WalletMessages } from '../../plugins/Wallet/messages'
+import { useI18N } from '../../utils/i18n-next-ui'
+import { useRemoteControlledDialog } from '../../utils/hooks/useRemoteControlledDialog'
+import { useChainId, useChainIdValid } from '../hooks/useChainState'
+import { resolveChainColor } from '../pipes'
+import { ChainId } from '../types'
+import { useValueRef } from '../../utils/hooks/useValueRef'
+import { currentSelectedWalletProviderSettings } from '../../plugins/Wallet/settings'
+
+const useStyles = makeStyles((theme: Theme) => {
+    return createStyles({
+        root: {},
+        providerIcon: {
+            fontSize: 18,
+            width: 18,
+            height: 18,
+        },
+        chainIcon: {
+            fontSize: 18,
+            width: 18,
+            height: 18,
+            marginLeft: theme.spacing(0.5),
+        },
+    })
+})
+
+export interface EthereumAccountButtonProps extends withClasses<KeysInferFromUseStyles<typeof useStyles>> {
```

Comment:
Hi as you can notice, `KeysInferFromUseStyles` has been deprecated because it is encouraging a bad coding style. Please explicitly note any class that you want to expose as a public API instead of inferring it from internal styles.